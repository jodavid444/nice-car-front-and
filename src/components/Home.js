import React, { useEffect, useRef, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getCars } from '../redux/cars/carsSlice';
import Cars from './cars/Cars';
import { Navbar } from '../Navbar/Navbar';

export function Home() {
  const cars = useSelector((state) => state.cars.cars) || [];
  const carsContainerRef = useRef(null);
  const dispatch = useDispatch();
  const [startIndex, setStartIndex] = useState(0);

  useEffect(() => {
    dispatch(getCars());
  }, [dispatch]);

  const scrollLeft = () => {
    if (carsContainerRef.current) {
      carsContainerRef.current.scrollBy({
        left: -carsContainerRef.current.offsetWidth,
        behavior: 'smooth',
      });
      setStartIndex((prevIndex) => Math.max(0, prevIndex - 1));
    }
  };

  const scrollRight = () => {
    if (carsContainerRef.current) {
      carsContainerRef.current.scrollBy({
        left: carsContainerRef.current.offsetWidth,
        behavior: 'smooth',
      });
      setStartIndex((prevIndex) => Math.min(prevIndex + 1, cars.length - 3));
    }
  };

  const visibleCars = cars.slice(startIndex, startIndex + 3).filter(Boolean);

  return (
    <div className="mainContainer">
      <div>
        <Navbar />
      </div>
      <div className="home-bg">
        <h2 className="home-header">Our Cars</h2>
        <h3 className="home-header two">Select your favorite car!</h3>
        <div className="arrows-cars">
          <button
            className="btn"
            onClick={scrollLeft}
            type="button"
          >
            Previous
          </button>
          <div className="carItems">
            <div className="carousel" ref={carsContainerRef}>
              <Cars cars={visibleCars} />
            </div>
          </div>

          <button
            className="btn next-btn"
            onClick={scrollRight}
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
  );
}

export default Home;
