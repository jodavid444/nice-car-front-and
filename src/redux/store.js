import { configureStore } from '@reduxjs/toolkit';
import carsReducer from './cars/carsSlice';
import reservationsReducer from './reservations/reservationsSlice';
import usersReducer from './users/usersSlice';

const store = configureStore({
  reducer: {
    cars: carsReducer,
    reservations: reservationsReducer,
    users: usersReducer,
  },
});
export default store;
