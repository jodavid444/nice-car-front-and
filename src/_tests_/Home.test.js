import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
// eslint-disable-next-line import/no-named-as-default
import Home from '../components/Home';
import store from '../redux/store';

describe('Home component', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
