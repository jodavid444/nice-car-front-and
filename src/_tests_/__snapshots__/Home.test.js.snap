// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match the snapshot 1`] = `
<div
  class="mainContainer"
>
  <div>
    <div
      class="headnavbar"
    >
      <button
        class="nav-btn nav-open-btn"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </button>
      <nav
        class="navbar"
      >
        <button
          class="nav-btn nav-close-btn"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 352 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            />
          </svg>
        </button>
        <img
          alt="Rent It icon"
          class="title"
          src="icon.jpg"
        />
        <ul
          class="navbarlinks"
        >
          <li>
            <a
              class="link"
              href="/home"
            >
              Cars
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/create-reservation"
            >
              Booking
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/reservations"
            >
              Reservations
            </a>
          </li>
          <li
            class="nan"
          >
            <a
              class="link"
              href="/add_car"
            >
              Add A Car
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/car-delete"
            >
              Delete A Car
            </a>
          </li>
        </ul>
        <h3
          class="navbar__footer"
        >
          Â©
           
          2023
        </h3>
      </nav>
    </div>
  </div>
  <div
    class="home-bg"
  >
    <h2
      class="home-header"
    >
      Our Cars
    </h2>
    <h3
      class="home-header two"
    >
      Select your favorite car!
    </h3>
    <div
      class="arrows-cars"
    >
      <button
        class="btn"
        type="button"
      >
        Previous
      </button>
      <div
        class="carItems"
      >
        <div
          class="carousel"
        >
          <div>
            No cars available
          </div>
        </div>
      </div>
      <button
        class="btn next-btn"
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
